content/reliability/docs/controlplane.md

```diff
@@ -174,6 +174,7 @@ AWS sets service limits (an upper limit on the number of each resource your team
 - Besides the limits from orchestration engines, there are limits in other AWS services, such as Elastic Load Balancing (ELB) and Amazon VPC, that may affect your application performance.
 - More about EC2 limits here: [EC2 service limits](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-resource-limits.html).
 - Each EC2 instance limits the number of packets that can be sent to the [Amazon-provided DNS server](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-dns.html#vpc-dns-limits) to a maximum of 1024 packets per second per network interface.
+- In EKS environment, etcd storage limit is 8GB as per [upstream guidance](https://etcd.io/docs/v3.5/dev-guide/limit/#storage-size-limit).Please monitor metric etcd_db_total_size_in_bytes to track etcd db size.
 
```

Comment:
@geetasg can you update the metric with the Prometheus query that you would use to monitor the size of the etcd database? The `etcd_db_total_size_in_bytes` returns multiple values (presumably for each node in the etcd cluster).

Reply:
"etcd_db_total_size_in_bytes" is exposed in `kube-apiserver` /metrics endpoint. Under the assumption that user configures `kube-apiserver` as a job, the query can be `max(etcd_db_total_size_in_bytes{job="kube-apiserver"} / (8 * 1024 * 1024 * 1024))`.

As of grafana alarm rule, I am referring to the upstream mixin example. 
1. etcdBackendQuotaLowSpace
2. etcdExcessiveDatabaseGrowth
```
          {
            alert: 'etcdBackendQuotaLowSpace',
            expr: |||
              (etcd_mvcc_db_total_size_in_bytes/etcd_server_quota_backend_bytes)*100 > 95
            ||| % $._config,
            'for': '10m',
            labels: {
              severity: 'critical',
            },
            annotations: {
              description: 'etcd cluster "{{ $labels.%s }}": database size exceeds the defined quota on etcd instance {{ $labels.instance }}, please defrag or increase the quota as the writes to etcd will be disabled when it is full.' % $._config.clusterLabel,
              summary: 'etcd cluster database is running full.',
            },
          },
          {
            alert: 'etcdExcessiveDatabaseGrowth',
            expr: |||
              increase(((etcd_mvcc_db_total_size_in_bytes/etcd_server_quota_backend_bytes)*100)[240m:1m]) > 50
            ||| % $._config,
            'for': '10m',
            labels: {
              severity: 'warning',
            },
            annotations: {
              description: 'etcd cluster "{{ $labels.%s }}": Observed surge in etcd writes leading to 50%% increase in database size over the past four hours on etcd instance {{ $labels.instance }}, please check as it might be disruptive.' % $._config.clusterLabel,
              summary: 'etcd cluster database growing very fast.',
            },
          },
```
ref. https://github.com/etcd-io/etcd/blob/main/contrib/mixin/mixin.libsonnet#L213-L240